pkg/etcd/etcd.go

```diff
@@ -632,34 +636,28 @@ func (e *ETCD) setSnapshotFunction(ctx context.Context) {
 // completion.
 func (e *ETCD) Restore(ctx context.Context) error {
 	// check the old etcd data dir
-	oldDataDir := etcdDBDir(e.config) + "-old"
-	if s, err := os.Stat(oldDataDir); err == nil && s.IsDir() {
-		logrus.Infof("Etcd already restored from a snapshot. Restart without --snapshot-restore-path flag. Backup and delete ${datadir}/server/db on each peer etcd server and rejoin the nodes")
-		os.Exit(0)
-	} else if os.IsNotExist(err) {
-		if e.config.ClusterResetRestorePath == "" {
-			return errors.New("no etcd restore path was specified")
-		}
-		// make sure snapshot exists before restoration
-		if _, err := os.Stat(e.config.ClusterResetRestorePath); err != nil {
-			return err
-		}
-		// move the data directory to a temp path
-		if err := os.Rename(etcdDBDir(e.config), oldDataDir); err != nil {
-			return err
-		}
-		sManager := snapshot.NewV3(nil)
-		if err := sManager.Restore(snapshot.RestoreConfig{
-			SnapshotPath:   e.config.ClusterResetRestorePath,
-			Name:           e.name,
-			OutputDataDir:  etcdDBDir(e.config),
-			OutputWALDir:   walDir(e.config),
-			PeerURLs:       []string{e.peerURL()},
-			InitialCluster: e.name + "=" + e.peerURL(),
-		}); err != nil {
-			return err
-		}
-	} else {
+	oldDataDir := etcdDBDir(e.config) + "-old-" + strconv.Itoa(int(time.Now().Unix()))
+	if e.config.ClusterResetRestorePath == "" {
```

Comment:
I wonder if we should place a check in at flag parse time to fail if this argument has no parameter.

Reply:
good idea, but let's do it as a separate post-ga follow up

Reply:
I agree, we can fix it in another pr, since this was part of the original restoration pr, its just indentation change in this pr